baybe/utils/interval.py

```diff
@@ -41,40 +45,66 @@ class Interval:
     """The upper end of the interval."""
 
     @upper.validator
-    def _validate_order(self, _: Any, value: float):  # noqa: DOC101, DOC103
+    def _validate_order(self, _: Any, upper: float) -> None:  # noqa: DOC101, DOC103
         """Validate the order of the interval bounds.
 
         Raises:
             ValueError: If the upper end is not larger than the lower end.
         """
-        if value <= self.lower:
+        if upper <= self.lower:
             raise ValueError(
-                f"The upper interval bound (provided value: {value}) must be larger "
+                f"The upper interval bound (provided value: {upper}) must be larger "
                 f"than the lower bound (provided value: {self.lower})."
             )
 
     @property
-    def is_finite(self):
+    def is_closed(self) -> bool:
+        """Check whether the interval is closed."""
+        return np.isfinite(self.lower) and np.isfinite(self.upper)
+
+    @property
+    def is_half_open(self) -> bool:
+        """Check whether the interval is half-open."""
+        return np.isfinite(self.lower) ^ np.isfinite(self.upper)
+
+    @property
+    def is_open(self) -> bool:
```

Comment:
See my comment at `is_closed` - using open here is *not* correct from a mathematical point.

Reply:
Should be fixed now. Please close if you agree